src/api/api.jl

```diff
@@ -1,16 +1,46 @@
 module API
 
-import Libdl
+using Libdl
 using Base: StringVector
+using Preferences: @load_preference
 
-const depsfile = joinpath(@__DIR__, "..", "..", "deps", "deps.jl")
-if isfile(depsfile)
-    include(depsfile)
+# We avoid calling Libdl.find_library to avoid possible segfault when calling
+# dlclose (#929).
+# The only difference with Libdl.find_library is that we allow custom dlopen
+# flags via the `flags` argument.
+function find_library_alt(libnames, extrapaths=String[]; flags=RTLD_LAZY)
+    for lib in libnames
+        for path in extrapaths
+            l = joinpath(path, lib)
+            p = dlopen(l, flags; throw_error=false)
+            if p !== nothing
+                dlclose(p)
+                return l
+            end
+        end
+        p = dlopen(lib, flags; throw_error=false)
+        if p !== nothing
+            dlclose(p)
+            return lib
+        end
+    end
+    return ""
+end
+
+const libpath = @load_preference("libhdf5path", nothing)
+if libpath === nothing
+    using HDF5_JLL
```

Comment:
```suggestion
    using HDF5_jll
```
Should fix the current CI issue

Reply:
Ok, I really missed something stupid :smile:. Thanks and done.